File: 105656458-main/first_try/sentimental-readability/readability.py
text = input("Text: ")
LETTERS = 0
WORDS = 1
SENTENCES = 0
for char in text:
    if char.isalpha():
        LETTERS += 1
    elif char == '.' or char == '?' or char == '!':
        SENTENCES += 1
    elif char == ' ':
        WORDS += 1
S = SENTENCES / WORDS * 100
L = LETTERS / WORDS * 100
index = round(0.0588 * L - 0.296 * S - 15.8)
if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")

File: 105656458-main/first_try/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include<math.h>
#include<ctype.h>
#include<string.h>

int count_letters(string text, int length);
int count_words(string text, int length);
int count_sentences(string text, int length);

int main(void)
{
    string text = get_string("Text:");
    int n = strlen(text);
    int letters = count_letters(text, n);
    int sentences = count_sentences(text, n);
    int words = count_words(text, n);
    float L = (float)letters / (float)words * 100;
    float S = (float)sentences / (float)words * 100;
    float x = 0.0588 * L - 0.296 * S - 15.8;
    int index = round(x);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if(index > 16)
    printf("Grade 16+\n");
    else
    {
        printf("Grade %i\n", index);
    }
}


int count_letters(string text, int length)
{
    int letters = 0;
    for(int i = 0; i < length; i++)
    {
        if(isalpha(text[i]))
        {
            letters++;
        }
    }
return letters;
}

int count_words(string text, int length)
{
    int words = 1;
    for(int i = 0; i < length;i++)
    {
        if(text[i] == 32)
        {
            words++;
        }
    }
    return words;
}

int count_sentences(string text, int length)
{
    int sentences = 0;
    for(int i = 0;i < length;i++)
    {
        if(text[i] == 33 || text[i] == 46 || text[i] == 63)
        {
            sentences++;
        }
    }
    return sentences;
}